### ISSUE TITLE
Autopublication failed to create versioned EML, causing resource list to be truncated

### ISSUE THREAD

**Issue by peterdesmet** [gbif/ipt#2063]
We just had a weird issue with our [INBO IPT](https://ipt.inbo.be/) (Version 2.7.3). We managed to fix it, but I'm posting it here to gain insights:

1. Homepage of IPT showed all 123 resources, but many were not accessible ("Resource not found")
2. The resources on my `Manage Resources` was reduced to 12, was expecting over a 100
3. Going to `Admin > IPT` settings showed a clearer error:
    > `florabank1-occurrences/eml-45.24.xml` could not be found
4. We manually created such a file and uploaded it to the server. This resolved the issue and all datasets were available again.

The root cause of the issue is not resolved however: `florabank1-occurrences` uses a 6-monthly auto publication. The latest publication resulted in:

- ✅ `dwca-45.24.zip` created
- ❌ `eml-45.24.xml` not created
- ❌ `florabank1-occurrences-45.23.rtf` not created

This is the publication log:

```
Archive generation started for version #45.24
Start writing data file for Darwin Core Occurrence
No lines were skipped due to errors for mapping Darwin Core Occurrence in source florabank1
No lines were skipped due to errors for mapping Darwin Core Occurrence in source florabank1
No lines with fewer columns than mapped for mapping Darwin Core Occurrence in source florabank1
All lines match the filter criteria for mapping Darwin Core Occurrence in source florabank1
Data file written for Darwin Core Occurrence with 5233549 records and 33 columns
All data files completed
EML file added
meta.xml archive descriptor written
Validating the core file: occurrence.txt. Depending on the number of records, this can take a while.
? Validating the core basisOfRecord is always present and its value matches the Darwin Core Type Vocabulary.
? Validating the core ID field occurrenceID is always present and unique.
No lines are missing occurrenceID
No lines have duplicate occurrenceID
✓ Validated each line has a occurrenceID, and each occurrenceID is unique
No lines are missing a basisOfRecord
All lines have basisOfRecord that matches the Darwin Core Type Vocabulary
No lines have ambiguous basisOfRecord 'occurrence'.
✓ Validated each line has a basisOfRecord, and each basisOfRecord matches the Darwin Core Type Vocabulary
Archive validated
Archive has been compressed
Archive version #45.24 generated successfully!
id certification path to requested target". ClientConnectionId:ae8817f2-49e8-463c-b8ce-52f0764e80b8
	at org.gbif.ipt.service.manage.impl.SourceManagerImpl.rowIterator(SourceManagerImpl.java:873)
	at org.gbif.ipt.task.GenerateDwca.dumpData(GenerateDwca.java:1175)
	... 8 more
```

@mike-podolskiy90

1. Any idea what might have caused `eml-45.24.xml` (and rtf) not be created?
2. Any way to prevent a missing EML to _not_ cause the resource list to be truncated? Note that the missing rtf file (which we did not recreate) does not cause this.

Ping @pietrH @niconoe

**Comment by mike-podolskiy90** [comment]
Thanks Peter. I'll have a look

**Comment by mike-podolskiy90** [comment]
#2025 should prevent resource list to be truncated.
I'm not sure what happened to the EML though. Can I have the whole stacktrace of the exception please?

**Comment by peterdesmet** [comment]
The exception was truncated in the `publication.log` so I can't retrieve more info.

**Comment by mike-podolskiy90** [comment]
It's really difficult to say the reason of failing. However I've released a new version (2.7.4) which has a better exception handling and which does not truncate resources. I think it'll be fine for now, otherwise we'll re-open it.
